src/main/java/org/kairosdb/datastore/cassandra/CassandraDatastore.java

```diff
@@ -813,10 +813,12 @@ private static void logQuery(DatastoreMetricQuery query,
                                  boolean limitExceeded,
                                  int limit,
                                  String index) {
-        final long endTime = Long.MAX_VALUE == query.getEndTime() ? System.currentTimeMillis() : query.getEndTime();
-        logger.warn("{}_query: uuid={} metric={} query={} read={} filtered={} start_time={} end_time={} duration={} exceeded={} limit={} index={}",
-                query.getQueryLoggingType(), query.getQueryUUID(), query.getName(), query.getTags(), rowReadCount, filteredRowKeys.size(),
-                query.getStartTime(), endTime, endTime - query.getStartTime(), limitExceeded, limit, index);
+        final boolean isUntilNow = Long.MAX_VALUE == query.getEndTime();
```

Comment:
this will never be true because of endTime normalization (see #119 )

Reply:
I can't see in that pull requests any changes to how `query.endTime` is defined. Why won't it work?

Reply:
`getEndTime` is populated in QueryParses here - https://github.com/zalando-zmon/kairosdb/pull/119/files#diff-dd7567379dfe9daac01e0b0aa6c86110R460

Reply:
`QueryParser` isn't a subclass of `QueryMetric`. [This `getEndTime` method](https://github.com/zalando-zmon/kairosdb/blob/1a57aa59e23661bed602fbf648fcdccb046ad2e8/src/main/java/org/kairosdb/core/datastore/QueryMetric.java#L122-L129) is used in case above.